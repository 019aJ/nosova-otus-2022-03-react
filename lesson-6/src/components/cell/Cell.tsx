import { Component } from "react"

interface CellProps {
  title?: string
  width?: string
  height?: string
  id?: string
}

interface CellState {
  title?: string
  width?: string
  height?: string
  id: string
}
export default class Cell extends Component<CellProps, CellState> {
  constructor(props: CellProps) {
    super(props)
    this.state = {
      title: props.title || "0",
      width: props.width || "50px",
      height: props.height || "50px",
      id: props.id || Math.floor(Math.random() * 1000).toString(),
    }
  }
  render() {
    return (
      <div
        id={this.state.id}
        style={{
          backgroundColor: "pink",
          width: this.state.width,
          height: this.state.height,
          border: "solid 1px black",
        }}
      >
        <div style={{ textAlign: "center", position: "relative", top: "30%" }}>
          {this.props.title}
        </div>
      </div>
    )
  }
}
