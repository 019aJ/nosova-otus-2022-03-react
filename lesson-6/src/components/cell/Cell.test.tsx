import Cell from "./Cell"
import { render } from "@testing-library/react"

describe("Cell snapshot tests", () => {
  it("base snapshot", () => {
    const { container } = render(<Cell id="197" />)
    expect(container).toMatchSnapshot()
  })

  it("some data snapshot", () => {
    const { container } = render(
      <Cell title="0" width="5px" height="5px" id="0" />
    )
    expect(container).toMatchSnapshot()
  })
})
