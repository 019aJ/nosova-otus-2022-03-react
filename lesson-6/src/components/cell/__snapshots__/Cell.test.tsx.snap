// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell snapshot tests base snapshot 1`] = `
<div>
  <div
    id="197"
    style="background-color: pink; width: 50px; height: 50px; border: 1px solid black;"
  >
    <div
      style="text-align: center; position: relative; top: 30%;"
    />
  </div>
</div>
`;

exports[`Cell snapshot tests some data snapshot 1`] = `
<div>
  <div
    id="0"
    style="background-color: pink; width: 5px; height: 5px; border: 1px solid black;"
  >
    <div
      style="text-align: center; position: relative; top: 30%;"
    >
      0
    </div>
  </div>
</div>
`;
